PowerShell Help for Microsoft.WinGet.Client (#4674)
microsoft/winget-cli: src/PowerShell/Help/Microsoft.WinGet.Client/Set-WinGetUserSettings.md

﻿---
external help file: Microsoft.WinGet.Client.Cmdlets.dll-Help.xml
Module Name: Microsoft.WinGet.Client
ms.date: 08/01/2024
online version:
schema: 2.0.0
title: Set-WinGetUserSettings
---

# Set-WinGetUserSettings

## SYNOPSIS

> Sets WinGet settings.

Reviewer: ```suggestion
Sets configuration settings of the WinGet client for the current user.
```

## SYNTAX

```
Set-WinGetUserSettings -UserSettings <Hashtable> [-Merge] [<CommonParameters>]
```

## DESCRIPTION

> Sets the behavior of various WinGet settings. For more information about WinGet settings, see
> [WinGet CLI Settings](https://aka.ms/winget-settings).

Reviewer: ```suggestion
This command sets configuration settings of the WinGet client for the current user. The user
settings file doesn't exist until you set a specific value. The file is stored in
`$env:LOCALAPPDATA\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\settings.json`.
For more information about WinGet settings, see
[WinGet CLI Settings](https://aka.ms/winget-settings).
```

## EXAMPLES

### Example 1: Set progress bar theme

```powershell

> Set-WinGetUserSettings -UserSettings @{ visual= @{ progressBar="rainbow"} }

Reviewer: ```suggestion
Set-WinGetUserSettings -UserSettings @{
    visual = @{
        progressBar = 'rainbow'
    }
}
```

```

Sets the theme of the progress bar to rainbow.

### Example 2: Merge install behavior settings

```powershell

> Set-WinGetUserSettings -UserSettings @{ installBehavior= @{ preferences= @{ scope = "user"}} } -Merge

Reviewer: ```suggestion
Set-WinGetUserSettings  -Merge -UserSettings @{
    installBehavior = @{
        preferences = @{
            scope = 'user'
        }
    }
}
```

```

Appends the user scope preference setting to the existing WinGet settings configuration.

> ## PARAMETERS

Reviewer: ````suggestion
### Example 3: Change multiple settings

```powershell
Set-WinGetUserSettings -UserSettings @{
    visual = @{
        progressBar = 'rainbow'
        anonymizeDisplayedPaths = $true
    }
}
```

## PARAMETERS
````

### -Merge

> Appends the provided UserSettings input to the existing settings configuration.

Reviewer: ```suggestion
By default, the command overwrites the current setting with the values provided. Use this parameter
to append the new settings to the existing configuration.
```
